src/Storages/MergeTree/ReplicatedMergeTreeQueue.cpp

```diff
@@ -221,6 +219,43 @@ void ReplicatedMergeTreeQueue::createLogEntriesToFetchBrokenParts()
     broken_parts_to_enqueue_fetches_on_loading.clear();
 }
 
+void ReplicatedMergeTreeQueue::addDropReplaceIntent(const MergeTreePartInfo & intent)
+{
+    std::lock_guard lock{state_mutex};
+    drop_replace_range_intents.push_back(intent);
+}
+
+void ReplicatedMergeTreeQueue::removeDropReplaceIntent(const MergeTreePartInfo & intent)
+{
+    std::lock_guard lock{state_mutex};
+    auto it = std::find(drop_replace_range_intents.begin(), drop_replace_range_intents.end(), intent);
```

Comment:
why not use https://en.cppreference.com/w/cpp/container/vector/erase2 and `chassert` on the return value?

Reply:
First of all I didn't know it exists, but as I checked it it erases all ellements, not just one. I want to remove only a single one.